lib/appsignal/utils.rb

```diff
@@ -14,12 +14,81 @@ def sanitize(params, only_top_level = false, key_sanitizer = nil)
     end
     extend ClassMethods
 
+    def self.data_generate(body)
+      Data.generate(body)
+    end
+
+    class Data
+      class << self
```

Comment:
Is there a reason this has to be a singleton class or can it be a module as well?